packages/vm/src/runTx.ts

```diff
@@ -440,6 +447,51 @@ async function _runTx(this: VM, opts: RunTxOpts): Promise<RunTxResult> {
     fromAccount.balance = BIGINT_0
   }
   await this.evm.journal.putAccount(caller, fromAccount)
+
+  const writtenAddresses = new Set<string>()
+  if (tx.supports(Capability.EIP7702EOACode)) {
+    const authorizationList = (<EIP7702CompatibleTx>tx).authorizationList
+    const MAGIC = new Uint8Array([5])
+    for (let i = 0; i < authorizationList.length; i++) {
+      const data = authorizationList[i]
+      const chainId = data[0]
+      const chainIdBN = bytesToBigInt(chainId)
+      if (chainIdBN !== BIGINT_0 && chainIdBN !== this.common.chainId()) {
+        // Chain id does not match, continue
+        continue
+      }
+      // Address to take code from
+      const address = data[1]
+      const nonceList = data[2]
+      const yParity = bytesToBigInt(data[3])
+      const r = data[4]
+      const s = data[5]
+
+      const rlpdSignedMessage = RLP.encode([chainId, address, nonceList])
+      const toSign = keccak256(concatBytes(MAGIC, rlpdSignedMessage))
+      const pubKey = ecrecover(toSign, yParity, r, s)
+      // Address to set code to
+      const authority = new Address(publicToAddress(pubKey))
+      const account = (await this.stateManager.getAccount(authority)) ?? new Account()
+
+      if (account.isContract()) {
+        // Note: this also checks if the code has already been set once
+        // So, if there are multiply entires for the same address, then this is fine
+        continue
+      }
+      if (nonceList.length !== 0 && account.nonce !== bytesToBigInt(nonceList[0])) {
+        continue
+      }
+
+      const addressConverted = new Address(address)
+      const addressCode = await this.stateManager.getContractCode(addressConverted)
+      await this.stateManager.putContractCode(authority, addressCode)
+
+      writtenAddresses.add(authority.toString())
```

Comment:
Don't we also need to check and see if the address provided in the authority tuple has already been set (or does the above `isContract` check take care of that)?

Reply:
Yes, this should be clear from the comment under `account.isContract()` but it is not super clear, have added some extra clarification